server/commands/list_objects.go

```diff
@@ -0,0 +1,234 @@
+)
+
+type ListObjectsQuery struct {
+	datastore                    storage.OpenFGADatastore
+	logger                       logger.Logger
+	tracer                       trace.Tracer
+	meter                        metric.Meter
+	listObjectsDeadlineInSeconds int
+	listObjectsMaxResults        uint32
+	resolveNodeLimit             uint32
+}
+
+// NewListObjectsQuery creates a ListObjectsQuery
+func NewListObjectsQuery(datastore storage.OpenFGADatastore, tracer trace.Tracer, logger logger.Logger, meter metric.Meter, listObjectsDeadlineInSeconds int, listObjectsMaxResults uint32, ResolveNodeLimit uint32) *ListObjectsQuery {
+	return &ListObjectsQuery{
+		datastore:                    datastore,
+		logger:                       logger,
+		tracer:                       tracer,
+		meter:                        meter,
+		listObjectsDeadlineInSeconds: listObjectsDeadlineInSeconds,
+		listObjectsMaxResults:        listObjectsMaxResults,
+		resolveNodeLimit:             ResolveNodeLimit,
+	}
+}
+
+// Execute the ListObjectsQuery, returning a list of object IDs
+func (q *ListObjectsQuery) Execute(ctx context.Context, req *openfgapb.ListObjectsRequest) (*openfgapb.ListObjectsResponse, error) {
+	err := q.validateInput(ctx, req.StoreId, req.Type, req.AuthorizationModelId, req.Relation)
+	if err != nil {
+		return nil, err
+	}
+
+	resultsChan := make(chan string, q.listObjectsMaxResults)
+	errChan := make(chan error)
+	resolvedChan := make(chan struct{})
+
+	timeoutCtx := ctx
+	if q.listObjectsDeadlineInSeconds != 0 {
+		var cancel context.CancelFunc
+		timeoutCtx, cancel = context.WithTimeout(ctx, time.Second*time.Duration(q.listObjectsDeadlineInSeconds))
+		defer cancel()
+	}
+
+	go func() {
+		q.performChecks(timeoutCtx, req.StoreId, req.AuthorizationModelId, req.Type, req.Relation, req.User, req.ContextualTuples, resultsChan, errChan, resolvedChan)
+	}()
+
+	select {
+	case <-timeoutCtx.Done():
+		return nil, serverErrors.NewInternalError("Timeout exceeded", timeoutCtx.Err())
+	case <-resolvedChan:
+		objectIDs := make([]string, 0)
+		for objectID := range resultsChan {
+			objectIDs = append(objectIDs, objectID)
+		}
+
+		return &openfgapb.ListObjectsResponse{
+			ObjectIds: objectIDs,
+		}, nil
+	case genericError := <-errChan:
+		return nil, genericError
+	}
+}
+
+func (q *ListObjectsQuery) ExecuteStreamed(ctx context.Context, req *openfgapb.StreamedListObjectsRequest, srv openfgapb.OpenFGAService_StreamedListObjectsServer) error {
+	err := q.validateInput(ctx, req.StoreId, req.Type, req.AuthorizationModelId, req.Relation)
+	if err != nil {
+		return err
+	}
+
+	resultsChan := make(chan string, q.listObjectsMaxResults)
+	errChan := make(chan error)
+	resolvedChan := make(chan struct{})
+
+	timeoutCtx := ctx
+	if q.listObjectsDeadlineInSeconds != 0 {
+		var cancel context.CancelFunc
+		timeoutCtx, cancel = context.WithTimeout(ctx, time.Second*time.Duration(q.listObjectsDeadlineInSeconds))
```

Comment:
Why do we use an integer just to turn it back into a Duration? Let's just use a duration type.